src/event_object_movement.c

```diff
@@ -217,175 +180,1045 @@ static u8 setup##_callback(struct ObjectEvent *objectEvent, struct Sprite *sprit
+};
+
+const u8 gAcroWheeliePedalDirectionAnimNums[] = {
+    [DIR_NONE]      = 0x20,
+    [DIR_SOUTH]     = 0x20,
+    [DIR_NORTH]     = 0x21,
+    [DIR_WEST]      = 0x22,
+    [DIR_EAST]      = 0x23,
+    [DIR_SOUTHWEST] = 0x20,
+    [DIR_SOUTHEAST] = 0x20,
+    [DIR_NORTHWEST] = 0x21,
+    [DIR_NORTHEAST] = 0x21,
+};
+
+const u8 gFishingDirectionAnimNums[] = {
+    [DIR_NONE]      = 0x24,
+    [DIR_SOUTH]     = 0x24,
+    [DIR_NORTH]     = 0x25,
+    [DIR_WEST]      = 0x26,
+    [DIR_EAST]      = 0x27,
+    [DIR_SOUTHWEST] = 0x24,
+    [DIR_SOUTHEAST] = 0x24,
+    [DIR_NORTHWEST] = 0x25,
+    [DIR_NORTHEAST] = 0x25,
+};
+
+const u8 gFishingNoCatchDirectionAnimNums[] = {
+    [DIR_NONE]      = 0x0,
+    [DIR_SOUTH]     = 0x0,
+    [DIR_NORTH]     = 0x1,
+    [DIR_WEST]      = 0x2,
+    [DIR_EAST]      = 0x3,
+    [DIR_SOUTHWEST] = 0x0,
+    [DIR_SOUTHEAST] = 0x0,
+    [DIR_NORTHWEST] = 0x1,
+    [DIR_NORTHEAST] = 0x1,
+};
+
+const u8 gFishingBiteDirectionAnimNums[] = {
+    [DIR_NONE]      = 0x4,
+    [DIR_SOUTH]     = 0x4,
+    [DIR_NORTH]     = 0x5,
+    [DIR_WEST]      = 0x6,
+    [DIR_EAST]      = 0x7,
+    [DIR_SOUTHWEST] = 0x4,
+    [DIR_SOUTHEAST] = 0x4,
+    [DIR_NORTHWEST] = 0x5,
+    [DIR_NORTHEAST] = 0x5,
+};
+
+const u8 gUnknown_83A648A[] = {
+    [DIR_NONE]      = 0x8,
+    [DIR_SOUTH]     = 0x8,
+    [DIR_NORTH]     = 0x9,
+    [DIR_WEST]      = 0xA,
+    [DIR_EAST]      = 0xB,
+    [DIR_SOUTHWEST] = 0x8,
+    [DIR_SOUTHEAST] = 0x8,
+    [DIR_NORTHWEST] = 0x9,
+    [DIR_NORTHEAST] = 0x9,
+};
+
+const u8 gRunningDirectionAnimNums[] = {
+    [DIR_NONE]      = 0x14,
+    [DIR_SOUTH]     = 0x14,
+    [DIR_NORTH]     = 0x15,
+    [DIR_WEST]      = 0x16,
+    [DIR_EAST]      = 0x17,
+    [DIR_SOUTHWEST] = 0x14,
+    [DIR_SOUTHEAST] = 0x14,
+    [DIR_NORTHWEST] = 0x15,
+    [DIR_NORTHEAST] = 0x15,
+};
+
+const u8 gTrainerFacingDirectionMovementTypes[] = {
+    [DIR_NONE]      = 0x08,
+    [DIR_SOUTH]     = 0x08,
+    [DIR_NORTH]     = 0x07,
+    [DIR_WEST]      = 0x09,
+    [DIR_EAST]      = 0x0A,
+    [DIR_SOUTHWEST] = 0x08,
+    [DIR_SOUTHEAST] = 0x08,
+    [DIR_NORTHWEST] = 0x07,
+    [DIR_NORTHEAST] = 0x07,
+};
+
+bool8 (*const gOppositeDirectionBlockedMetatileFuncs[])(u8) = {
+    MetatileBehavior_IsSouthBlocked,
+    MetatileBehavior_IsNorthBlocked,
+    MetatileBehavior_IsWestBlocked,
+    MetatileBehavior_IsEastBlocked
+};
+
+bool8 (*const gDirectionBlockedMetatileFuncs[])(u8) = {
+    MetatileBehavior_IsNorthBlocked,
+    MetatileBehavior_IsSouthBlocked,
+    MetatileBehavior_IsEastBlocked,
+    MetatileBehavior_IsWestBlocked
+};
+
+static const struct Coords16 sDirectionToVectors[] = {
+    [DIR_NONE]      = { 0,  0},
+    [DIR_SOUTH]     = { 0,  1},
+    [DIR_NORTH]     = { 0, -1},
+    [DIR_WEST]      = {-1,  0},
+    [DIR_EAST]      = { 1,  0},
+    [DIR_SOUTHWEST] = {-1,  1},
+    [DIR_SOUTHEAST] = { 1,  1},
+    [DIR_NORTHWEST] = {-1, -1},
+    [DIR_NORTHEAST] = { 1, -1},
+};
+
+const u8 gFaceDirectionMovementActions[] = {
```

Comment:
Use the movement action constants here (931 and below)

Reply:
Resolved